src/Search_Replace_Command.php

```diff
@@ -165,10 +168,14 @@ public function __invoke( $args, $assoc_args ) {
 		$total           = 0;
 		$report          = array();
 		$this->dry_run         = \WP_CLI\Utils\get_flag_value( $assoc_args, 'dry-run' );
-		$php_only        = \WP_CLI\Utils\get_flag_value( $assoc_args, 'precise' );
+		$php_only              = \WP_CLI\Utils\get_flag_value( $assoc_args, 'precise' );
 		$this->recurse_objects = \WP_CLI\Utils\get_flag_value( $assoc_args, 'recurse-objects', true );
 		$this->verbose         =  \WP_CLI\Utils\get_flag_value( $assoc_args, 'verbose' );
 		$this->format          = \WP_CLI\Utils\get_flag_value( $assoc_args, 'format' );
+		$this->regex_limit    = \WP_CLI\Utils\get_flag_value( $assoc_args, 'regex-limit', -1 );
+		if ( 0 === intval( $this->regex_limit ) ) {
```

Comment:
Doing `(int) $this->regex_limit` is the preferred modern way of doing this. Also not checking for negative numbers. Maybe (in the following `if ( ( $this->regex = ...` block) do explicitly:
```
if ( null !== ( $regex_limit = \WP_CLI\Utils\get_flag_value( $assoc_args, 'regex-limit' ) ) ) {
		if ( ! preg_match( '/^[0-9]+$/', $regex_limit ) || 0 === (int) $regex_limit ) {
			WP_CLI::error( '`--regex-limit` expects a non-zero positive integer.' );
		}
		$this->regex_limit = (int) $regex_limit;
	}
}
```
Note leaving out the default on `get_flag_value()` so can check for `null` - see above default for private var.

Reply:
Actually `0` can be passed and will work as expected, i.e. won't replace anything.

Reply:
Would that be useful?!

Reply:
Not necessarily but it's consistent with the behaviour of `preg_replace`.